src/stcal/ramp_fitting/ols_fit.py

```diff
@@ -363,174 +149,418 @@ def ols_ramp_fit_sliced(
 
-    groupgap : int
-        The number of frames that are not included in the group average
+    pool = Pool(processes=number_slices)
+    pool_results = pool.starmap(ols_ramp_fit_single, slices)
+    pool.close()
+    pool.join()
 
-    nframes : int
-        The number of frames that are included in the group average
+    # Reassemble results
+    image_info, integ_info, opt_info = assemble_pool_results(
+        ramp_data, save_opt, pool_results, rows_per_slice)
 
-    dropframes1 :
-        The number of frames dropped at the beginning of every integration
+    return image_info, integ_info, opt_info
 
-    int_times : None
-        Not used
 
-    Returns
-    -------
-    new_model.data : ndarray
-        The output final rate of each pixel, 2-D float
+def assemble_pool_results(ramp_data, save_opt, pool_results, rows_per_slice):
+    """
+    Takes the list of results from the starmap pool method and assembles the
+    slices into primary tuples to be returned by `ramp_fit`.
 
-    new_model.dq : ndarray
-        The output pixel dq for each pixel, 2-D flag
+    Parameters
+    ----------
+    ramp_data: RampData
+        The data needed for ramp fitting.
 
-    new_model.var_poisson : ndarray
-        The variance in each pixel due to Poisson noise, 2-D float32
+    save_opt: bool
+        The option to save the optional results.
 
-    new_model.var_rnoise : ndarray
-        The variance in each pixel due to read noise, 2-D float32
+    pool_results: list
+        The list of return values from ols_ramp_fit_single for each slice.
 
-    new_model.err : ndarray
-        The output total variance for each pixel, 2-D float32
+    rows_per_slice: list
+        The number of rows in each row.
 
-    int_data : ndarray
-        The rate for each pixel in each integration, 3-D float32
+    Return
+    ------
+    image_info: tuple
+        The tuple of computed ramp fitting arrays.
 
-    int_dq : ndarray
-        The pixel dq flag for each integration, 3-D float32
+    integ_info: tuple
+        The tuple of computed integration fitting arrays.
+
+    opt_info: tuple
+        The tuple of computed optional results arrays for fitting.
+    """
+    # Create output arrays for each output tuple.  The input ramp data and
+    # slices are needed for this.
+    image_info, integ_info, opt_info = create_output_info(
+        ramp_data, pool_results, save_opt)
+
+    # Loop over the slices and assemble each slice into the main return arrays.
+    current_row_start = 0
+    for k, result in enumerate(pool_results):
+        image_slice, integ_slice, opt_slice = result
+        nrows = rows_per_slice[k]
+
+        get_image_slice(image_info, image_slice, current_row_start, nrows)
+        get_integ_slice(integ_info, integ_slice, current_row_start, nrows)
+        if save_opt:
+            get_opt_slice(opt_info, opt_slice, current_row_start, nrows)
+        current_row_start = current_row_start + nrows
+
```

Comment:
Does get_integ_slice(  ) assemble the output info from the slices whose output info was assembled from each image by get_image_slice(  ) ?  Maybe add a bit more documentation to clarify what these functions do.

Reply:
No.  The sliced (by row) return values are returned from `starmap` where each sliced input is used as parameters for `ols_ramp_fit_single`.  These sliced return values are used by the `get_xxxx_slice()` functions to assemble each slice into a single `xxxx` product to be returned from ramp fitting.